Userland/Libraries/LibGfx/AntiAliasingPainter.cpp

```diff
@@ -169,3 +170,218 @@ void Gfx::AntiAliasingPainter::draw_cubic_bezier_curve(const FloatPoint& control
         draw_line(fp1, fp2, color, thickness, style);
     });
 }
+
+void Gfx::AntiAliasingPainter::draw_circle(IntPoint center, int radius, Color color)
+{
+    /*
+    Algorithm from: https://cs.uwaterloo.ca/research/tr/1984/CS-84-38.pdf
+    Inline comments are from the paper.
+    */
+
+    // TODO: Generalize to ellipses (see paper)
+
+    constexpr int subpixel_resoulution = 255;
+    const int intensity { color.alpha() };
+
+    // Note: Variable names below are from the paper
+
+    /* Current pixel address */
+    int i { 0 }, q { radius };
```

Comment:
It feels weird to be initializing integers this way. Perhaps just:
```cpp
int i = 0, q = radius;
```
would be nicer. Same for the other initializations below

Reply:
Perhaps just split them into multiple lines instead?

Reply:
Got a bit carried away with with the C++ initializers :sweat_smile: